pkg/services/vimmachine.go

```diff
@@ -116,46 +120,48 @@ func (v *VimMachineService) SyncFailureReason(machineCtx capvcontext.MachineCont
 }
 
 // ReconcileNormal reconciles create and update events for the VSphere VM.
-func (v *VimMachineService) ReconcileNormal(machineCtx capvcontext.MachineContext) (bool, error) {
+func (v *VimMachineService) ReconcileNormal(ctx context.Context, machineCtx capvcontext.MachineContext) (bool, error) {
+	log := ctrl.LoggerFrom(ctx).WithValues("VSphereMachine", klog.KRef(machineCtx.GetVSphereMachine().GetNamespace(), machineCtx.GetVSphereMachine().GetName()))
+	ctx = ctrl.LoggerInto(ctx, log)
```

Comment:
Same as for vmop, aren't these kv pairs already there from the calling function? If no, let's add them there and remove here. (also at ReconcileDelete)
```suggestion
	log := ctrl.LoggerFrom(ctx)
```